src/chat/model_selector_item.rs

```diff
@@ -126,9 +126,9 @@ impl ModelSelectorItemRef {
         inner.entity = ModelSelectorEntity::Model(model);
     }
 
-    pub fn set_agent(&mut self, agent: MofaAgent) {
+    pub fn set_agent(&mut self, agent: &MofaAgent) {
         let Some(mut inner) = self.borrow_mut() else { return };
-        inner.entity = ModelSelectorEntity::Agent(agent);
+        inner.entity = ModelSelectorEntity::Agent(agent.clone());
```

Comment:
You want to work with an owned `MofaAgent`, so keep taking an owned `MofaAgent` :P

I would avoid cloning inside the function if possible and let the caller clone if it's really necessary.

Does inverting the two last lines work?

```rust
pub fn set_agent(&mut self, agent: MofaAgent) {
    let Some(mut inner) = self.borrow_mut() else {
        return;
    };
    inner.chat_agent_avatar(id!(avatar)).set_agent(&agent);
    inner.entity = ModelSelectorEntity::Agent(agent);
}
```